### ISSUE TITLE
 keras.metrics.Accuracy != keras.metrics.accuracy

### ISSUE THREAD

**Issue by rodrigob** [tensorflow/tensorflow#32761]
The following test 
```
@test_util.run_all_in_graph_and_eager_modes
class KerasAccuracyTest(test.TestCase):

 def test_accuracy_vs_accuracy(self):
    y_true = constant_op.constant([1, 0, 1])
    y_pred = constant_op.constant([0.8, 0.1, 0.9])
    ret_a_tensor = metrics.accuracy(y_true, y_pred)
    ret_a = np.mean(self.evaluate(ret_a_tensor))
    
    acc_obj = metrics.Accuracy(name='my_acc')
    self.evaluate(variables.variables_initializer(acc_obj.variables))
    update_op = acc_obj.update_state(y_true, y_pred)
    self.evaluate(update_op)
    ret_b = self.evaluate(acc_obj.result())
    self.assertEqual(ret_a, ret_b)
```

does not pass because 
https://github.com/tensorflow/tensorflow/blob/3d5e79e08ae299812e0eaf6183f4886591e932bd/tensorflow/python/keras/metrics.py#L576-L577

seems to be doing two casts they should not.

Also it is unclear to me what is the expected value for `ret_a` (which I would have understood to be 1, but currently returns 1/3).

**Comment by arshiaadouli** [comment]
yes right

**Comment by arshiaadouli** [comment]
no wrong

**Comment by sushreebarsa** [comment]
@rodrigob  Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999)

**Comment by tensorflowbutler** [comment]
Hi There,

 This is a stale issue. As you are using an older version of tensorflow, we are checking to see if you still need help on this issue. Please test the issue with the latest TensorFlow (TF2.7 and tf-nightly). If the issue still persists with the newer versions of TF, please feel free to open it in [keras-team/keras](https://github.com/keras-team/keras/issues) repository by providing details about the issue and a standalone code to reproduce the issue. Thanks! 

 Please note that Keras development has moved to a separate Keras-team/keras repository to focus entirely on only Keras. Thanks!
